tools/serialize_nano_core/src/main.rs

```diff
@@ -0,0 +1,30 @@
+//! Tool that creates a serialized representation of the symbols in the `nano_core` object file.
+
+mod parse;
+
+use mod_mgmt::serde::SerializedCrate;
+use std::io::Write;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let arg = &std::env::args().collect::<Vec<String>>()[1];
+    let str = std::fs::read_to_string(arg)?;
```

Comment:
i'm surprised this actually compiles, since `str` is a reserved keyword. Let's change it to avoid potential future compiler errors.